### ISSUE TITLE
Erlang Timeout from non-configurable Constant

### ISSUE THREAD

**Issue by andercs** [KronicDeth/intellij-elixir#2715]
**Describe the bug**
Pretty much the same issue as #1932 but I'm willing to help troubleshoot where I can. The weird thing is that it only seems to be happening when I'm on Zoom calls with other people. As soon as I jump off the call, then the Debugger fires right up and if I sit in a Zoom call by myself, then it still starts without issue.

Wondering if it's a race condition for the debugger starting and trying to connect and the vm compiling everything and starting the Phoenix's Cowboy server. Possibly that streaming multiple other people barely ekes out in favor of the Timeout instead of the Debugger successfully connecting.

(By the way, I'm just getting started with Elixir for the first time ever at a new job and love this plugin over the experience in VS Code. You do awesome work @KronicDeth! Thank you for using your personal time to do the Elixir community this huge service.)

**To Reproduce**
1. Open Zoom
2. Try to demo the Elixir debugger to at least one other co-worker
3. Enjoy an IDE error with `OtpErlangExit timeout`
4. Apologize to co-worker
5. Hop off call and turn on the debugger only to watch it all compile and connect

**Expected behavior**
Increasing the debugger timeout period?

**Desktop:**
 - OS: macOS
 - Version 12.4

**Erlang:**
- Installer `asdf`
- Version 23.1.1

**Elixir:**
- Installer `asdf`
- Version 1.11.0-otp-23

**Plugin:**
- Version 13.1.0

**Additional context**

**Comment by andercs** [comment]
It consistently doesn't work on a Zoom call regardless of the number of retries, and I think maybe once it didn't start off the call and restarting it like you suggest got it working again.

If the answer for now is just to start the Debugger before the Zoom call, then we can roll with that.

**Comment by KronicDeth** [comment]
Yeah, I would just start the debugger before the Zoom call, while I use Zoom for client meetings and even pairing I've never had this happen, so it's so weird and there's a work-around that this will be low priority.

**Comment by andercs** [comment]
@KronicDeth actually was going to ask, since it's a timeout issue, I did see org.elixir_lang.debugger.Node on Line 255 seemed to be the place where the Timeout period was set as a constant (TIMEOUT_IN_MILLISECONDS) equal to 1 minute. Is there an easy way to make that value configurable from the Intellij UI to test setting it to something longer than a minute?

**Comment by andercs** [comment]
@KronicDeth any thoughts on my last comment? Not super familiar with Intellij plugins creating custom run configurations, but I'm guessing one can modify https://github.com/KronicDeth/intellij-elixir/blob/main/src/org/elixir_lang/mix/Configuration.kt and other files that reference it to introduce a new line in the run config for "Elixir Mix"?

**Comment by KronicDeth** [comment]
Since it’s just for the debugging I would put it on the same tab as the interpreted module filters.  You’ll need to edit the configuration for how it is stored and the form for the UI controls

**Comment by sh41** [comment]
Confirmed, and there is a second half to it that hasn't been described here yet.

**The constant.** Every request the IDE makes of the debugged node — attach, interpret, set and remove
breakpoint, evaluate, each stepping operation — goes through one call wrapped in a hard-coded 60 second
timeout. It has held that value since the current debugger landed in 2018 and has never been made
configurable: the plugin's debugger settings expose exactly one thing, the stepping module filters, and
there is no timeout field anywhere. When it expires you get `OtpErlangExit: timeout`, which is the title
of every report in this family.

**The part that turns a slow session into a dead one.** Session start-up runs its steps in order and
abandons the rest if one raises. `interpret` is queued *before* `attach`. So on a project where
interpreting the code path takes longer than sixty seconds, the timeout fires on `interpret`, the
remaining steps are abandoned, and `attach` never runs at all. Meanwhile the debugged node is sitting in
a receive with no timeout of its own, waiting for exactly that attach, and will sit there indefinitely
printing nothing further. That is why this can present as a hang rather than as an error, and it is
almost certainly the same thing being reported in #3084.

Worth noting that node calls were moved onto a single serialised dispatcher in April, which makes
queueing behind one slow call more likely than it was before, not less.

So there are two distinct changes here, and they are independent: making the timeout configurable — or
simply not applying an interpret-sized timeout to `attach` — and not abandoning `attach` when an earlier
step times out.

I'm closing #1537, #1932 and #3339 as duplicates of this one. This thread has the clearest diagnosis of
the four, and this issue is now tracked under #3936 with the rest of the debugger work.

This issue was looked at as part of an AI-assisted review of the backlog, and a human checked this
reply before it was posted. If any of it is wrong, please say so here.
